### ISSUE TITLE
Warning: Cannot call method 'split' of undefined Use

### ISSUE THREAD

**Issue by tiemay** [AdesisNetlife/grunt-jade-i18n#22]
I'm having trouble running this module:
```
$ grunt
Running "jade:templates" (jade) task
Warning: Cannot call method 'split' of undefined Use --force to continue.

Aborted due to warnings.
```
I even copy pasted the documentation examples to try to get it to work:
Gruntfile.js:
```
module.exports = function(grunt) {

    // Project configuration.
    grunt.initConfig({
      jade: {
        templates: {
          options: {
            // jade i18n specific options
            i18n: {
              locales: 'locales/*.json',
              namespace: '$i18n'
            },
            // Jade specific options
            pretty: true
          },
          files: {
            "public/index.html": ["views/index.jade"]
          }
        }
      }
    });

    // Load plugins for the tasks.
    grunt.loadNpmTasks('grunt-jade-i18n');

    // Default task(s).
    grunt.registerTask('default', ['jade']);

};
```
locales/es_ES.json:
```
{
  "hello": {
    "world": "Hola Mundo"
  }
}
```
views/index.jade:
```
body
  h1 #{$i18n.hello.world}!
  p Using locale #{$localeName}
```

But I'm still getting that error. Here's my grunt version info:
```
$ grunt --version
grunt-cli v0.1.13
grunt v0.4.5
```
I'm using `"grunt-jade-i18n": "~0.3.2"` as a devDependency in package.json, and doing `grunt --force` doesn't help.

**Comment by h2non** [comment]
Could you provide the error stack trace? Run:
```
grunt jade --verbose
```

**Comment by tiemay** [comment]
Oh weird, doing `grunt jade` with or without `--verbose` works as expected, the problem only occurs when I do plain `grunt`:
```
$ grunt
Running "jade:templates" (jade) task
Warning: Cannot call method 'split' of undefined Use --force to continue.

Aborted due to warnings.


$ grunt jade
Running "jade:templates" (jade) task
>> Loading locale 'es_ES'

Running "contrib-jade:templates-es_ES" (contrib-jade) task
>> 1 file created.

Done, without errors.


$ grunt jade --verbose
Initializing
Command-line options: --verbose

Reading "Gruntfile.js" Gruntfile...OK

Registering Gruntfile tasks.
Initializing config...OK

Registering "grunt-jade-i18n" local Npm module tasks.
Reading /media/user/storage/tmp/workspace/i18n/node_modules/grunt-jade-i18n/package.json...OK
Parsing /media/user/storage/tmp/workspace/i18n/node_modules/grunt-jade-i18n/package.json...OK

Registering "/media/user/storage/tmp/workspace/i18n/node_modules/grunt-jade-i18n/node_modules/grunt-contrib-jade/tasks" tasks.
Loading "jade.js" tasks...OK
+ jade
Loading "i18n.coffee" tasks...OK
- jade
+ contrib-jade, jade
Loading "Gruntfile.js" tasks...OK
+ default

Running tasks: jade

Running "jade" task

Running "jade:templates" (jade) task
Verifying property jade.templates exists in config...OK
Files: views/index.jade -> public/index.html
Options: i18n={"locales":"locales/*.json","namespace":"$i18n"}, pretty
Reading locales/es_ES.json...OK
Reading /media/user/storage/tmp/workspace/i18n/node_modules/grunt-jade-i18n/tasks/temp/templates/es_ES.json...OK
>> Loading locale 'es_ES'
Reading translation data: locales/es_ES.json
Reading locales/es_ES.json...OK
Parsing locales/es_ES.json...OK
Reading locales/es_ES.json...OK
Parsing locales/es_ES.json...OK

Running "contrib-jade" task

Running "contrib-jade:templates-es_ES" (contrib-jade) task
Verifying property contrib-jade.templates-es_ES exists in config...OK
Files: views/index.jade -> public/es_ES/index.html
Options: namespace="JST", separator="\n\n", amd=false, i18n={"locales":"locales/*.json","namespace":"$i18n"}, pretty, data={"hello":{"world":"Hola Mundo"},"$i18n":{"hello":{"world":"Hola Mundo"}},"$localeName":"es_ES"}
Reading views/index.jade...OK
Writing public/es_ES/index.html...OK
File public/es_ES/index.html created.
>> 1 file created.

Done, without errors.
```
Am I doing something wrong with `grunt.registerTask('default', ['jade']);` in Gruntfile.js? It looks right as far as I can tell.

**Comment by tiemay** [comment]
`grunt watch` and `grunt server` also work ok.

**Comment by h2non** [comment]
Grunt runs the task, even if the flag `--verbose` is passed or not. All points that your trouble is agnostic of the task implementation, since it has been used for a while with the same config and all is working as expected.

**Comment by h2non** [comment]
Please, check you have defined the proper config object, and doesn't have conflict with other tasks and the file paths and directories exists in your project

**Comment by tiemay** [comment]
The files are there
```
$ tree -I node_modules
.
├── Gruntfile.js
├── locales/
│   └── es_ES.json
├── package.json
├── public/
│   └── es_ES/
└── views/
    └── index.jade
```
Jade is the only task I'm defining, and it's configuration is copied nearly verbatim from the documentation. I tried a couple other ways of configuring it to, but none helped. Are you not able to reproduce the error?

**Comment by 7wonders** [comment]
Im running into similar problems using command "grunt" or "grunt jade".

grunt --verbose

´´´
Running "jade:templates" (jade) task
Verifying property jade.templates exists in config...OK
Files: src/index.jade -> out/index.html
Warning: Cannot call method 'split' of undefined Use --force to continue.

Aborted due to warnings.
´´´

grunt jade --verbose
´´´
Loading "jade.js" tasks...OK
+ jade
Loading "i18n.coffee" tasks...OK
- jade
+ contrib-jade, jade
Initializing config...OK
Loading "Gruntfile.js" tasks...OK
+ default

Running tasks: jade

Running "jade" task

Running "jade:templates" (jade) task
Verifying property jade.templates exists in config...OK
Files: src/index.jade -> out/index.html
Options: i18n={"locales":"locales/*.json","namespace":"$i18n"}, pretty=false

Running "contrib-jade" task
>> No "contrib-jade" targets found.
Warning: Task "contrib-jade" failed. Use --force to continue.

Aborted due to warnings.
```

No files are being generated :(

**Comment by 7wonders** [comment]
did a prune, reinstall and sorted all my tasks out. The error that this issue is for only seems to happen when running as part of default task but I think is related to the other error anyway. I have created issue #23 just in case they are separate.

**Comment by h2non** [comment]
I've fixed this error. This is caused when running multiple tasks in Grunt, since the tasks tries to discover those tasks to avoid possible conflicts.

I've just released a new fresh `0.3.4` version that fixes this issue.
```bash
npm update grunt-jade-i18n
```

I'm sorry for the troubles, and thank you for reporting the issue
